### PR TITLE
Feature/lp hourly summarizer

### PR DESCRIPTION
**Pull request by MessaoudAzidane**
### Summary of changes:
- Added hourlySummarizer class with all summarizations as methods
- Added unit tests for the summarization methods.
- Added sql docs.

### Steps for testing: 
- You should have already some test data from the previous services (aggregator) 
- Create `hourly_data` table by running this SQL DDL query 
```
CREATE TABLE hourly_data (
    id SERIAL PRIMARY KEY,
    address VARCHAR(100) NOT NULL,
    open_timestamp_utc TIMESTAMP WITH TIME ZONE NOT NULL,
    close_timestamp_utc TIMESTAMP WITH TIME ZONE NOT NULL,
    close_reserves_0 DOUBLE PRECISION NOT NULL,
    close_reserves_1 DOUBLE PRECISION NOT NULL,
    num_swaps_0 BIGINT NOT NULL,
    num_swaps_1 BIGINT NOT NULL,
    num_mints BIGINT NOT NULL,
    num_burns BIGINT NOT NULL,
    mints_0 DOUBLE PRECISION NOT NULL,
    mints_1 DOUBLE PRECISION NOT NULL,
    burns_0 DOUBLE PRECISION NOT NULL,
    burns_1 DOUBLE PRECISION NOT NULL,
    volume_0 DOUBLE PRECISION NOT NULL,
    volume_1 DOUBLE PRECISION NOT NULL,
    max_block INTEGER NOT NULL,
    close_lp_token_supply NUMERIC(155, 0) NOT NULL
);
```
- Run the main.py:
`python main.py`

### How to run unit tests:
`python -m unittest tests/test_hourly_summarizer.py`

### CODE REVIEW

File: cspr_summarization/services/lp_hourly_summarizer.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+from cspr_summarization.entities.Blocks import *
```
Comment (AhmadPhazeRo) [inline cspr_summarization/services/lp_hourly_summarizer.py:1]: name the imported object

File: cspr_summarization/services/lp_hourly_summarizer.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import numpy as np
+from django.db.models import F
+from decimal import Decimal
+
+class LpHourlySummarizer:
+  
+  def __init__(self, start_hour, end_hour):
+    self.start_hour = start_hour
+    self.end_hour = end_hour
+    blocks = Blocks.objects.filter(timestamp_utc__range=(self.start_hour, self.end_hour)).values('block_number', 'timestamp_utc').order_by('-timestamp_utc')
```
Comment (AhmadPhazeRo) [inline cspr_summarization/services/lp_hourly_summarizer.py:21]: split to seperate lines

File: cspr_summarization/services/lp_hourly_summarizer.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    self.last_hour_block_numbers = pd.DataFrame.from_records(blocks)
+    all_pairs = AllPairs.objects.values('id', 'contract_address', 'token0_decimals', 'token1_decimals', 'token0_address', 'token1_address')
+    self.all_pairs = pd.DataFrame.from_records(all_pairs)
+  
+  '''
+  # create a record for each pair, with all values at zero
+  '''
+  def init_hourly_data(self):
+    for index, pair in self.all_pairs.iterrows():
```
Comment (AhmadPhazeRo) [inline cspr_summarization/services/lp_hourly_summarizer.py:30]: suggestion (not required): can do this in parallel

**Review by AhmadPhazeRo — CHANGES_REQUESTED** [review]
a few simple modifications. Thank you

**Review by AhmadPhazeRo — APPROVED** [review]

**Review by AhmadPhazeRo — APPROVED** [review]
code style LGTM

**Review by specialistvlad — APPROVED** [review]

**Review by AhmadPhazeRo — COMMENTED** [review]
Stylistically LGTM but have not run yet

**Comment by AhmadPhazeRo** [comment]
suggestion, but not required, since this is the best time to use the following recommendations given the code is in it's initial phase:
[black ](https://black.readthedocs.io/en/stable/getting_started.html) for formatting code (simple pip install, DON'T add to requirements though since this is for your own usage)
[isort ](https://pycqa.github.io/isort/) for sorting the imports within the file(s) (simple pip install, DON'T add to requirements though since this is for your own usage)

**Comment by specialistvlad** [comment]
Closing because #6 contains these changes

**Pull request closed without merging**